entropy/coin_flip/app/src/flip_coin.ts

```diff
@@ -29,75 +45,80 @@ const parser = yargs(hideBin(process.argv))
 
 async function main() {
   const argv = await parser.argv;
+  const chainName = argv.chainName;
+  if (chainName !== "optimism-sepolia" && chainName !== "arbitrum-sepolia") {
+    throw new Error("Invalid chain name");
+  }
+
+  let chain: Chain;
+  if (chainName === "optimism-sepolia") {
+    chain = optimismSepolia;
+  } else {
+    chain = arbitrumSepolia;
+  }
+
+  const client = createWalletClient({
+    chain: chain,
+    account: privateKeyToAccount(argv.privateKey as Hex),
```

Comment:
Is there a hex parser somewhere?  It's always better to parse / validate than it is to cast -- with casting code like this it's very very easy to trigger a state where the type system says that the value is a particular type (`Hex`), but at runtime it's actually a different type (e.g. I can just pass a non-hex string and the type checker will be invalid).  This kind of thing leads to subtle bugs over time.

Reply:
I agree, but I couldn't find a way to do right now.

Reply:
You probably want https://viem.sh/docs/utilities/isHex#ishex

```ts
import { isHex } from 'viem';

const main = () => {
    ...
    if (!isHex(argv.privateKey)) {
        throw new InvalidPrivateKeyError(argv.privateKey);
    }
    ...
};

class InvalidPrivateKeyError extends Error {
    override constructor(privateKey: string) {
        super(`Invalid private key: "${privateKey}" is not a valid hex string`);
    }
}
```